aten/src/ATen/native/cpu/MaxPoolKernel.cpp

```diff
@@ -471,7 +481,7 @@ void max_pool2d_kernel_impl(
       break;
     }
     case at::MemoryFormat::ChannelsLast: {
-      AT_DISPATCH_FLOATING_TYPES_AND(ScalarType::BFloat16, input.scalar_type(), "max_pool2d_channels_last", [&] {
+      AT_DISPATCH_ALL_TYPES_AND(ScalarType::BFloat16, input.scalar_type(), "max_pool2d_channels_last", [&] {
```

Comment:
is complex part of ALL_TYPES?

Reply:
no, all_types == floating_point types + integral types: https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/Dispatch.h#L378